test_communication/CMakeLists.txt

```diff
@@ -127,6 +127,32 @@ if(BUILD_TESTING)
     )
   endfunction()
 
+  function(custom_security_test_c target)
+    set(SKIP_TEST "")
+    ament_add_gtest(
+      "${target}${target_suffix}" ${ARGN}
+      TIMEOUT 90
+      ${SKIP_TEST}
+      APPEND_LIBRARY_DIRS "${append_library_dirs}"
+      ENV
+      RCL_ASSERT_RMW_ID_MATCHES=${rmw_implementation}
+      RMW_IMPLEMENTATION=${rmw_implementation}
+      ROS_SECURE_ROOT="${VALID_ROS_SECURE_ROOT}"
+    )
+    target_link_libraries(${target}${target_suffix}
```

Comment:
Please see the docblock for the function: https://github.com/ament/ament_cmake/blob/dc243d84429ccefbe3532c59a242a114eb20bad6/ament_cmake_gtest/cmake/ament_add_gtest.cmake#L21-L22

You need this check (other existing code might need it too):

```
if(TARGET ${target}${target_suffix})
   ...
endif()
```